test/handlers/fee/dynamic/collectFeeGenericEVM.js

```diff
@@ -0,0 +1,401 @@
+      );
+    });
+    const balanceAfter = await web3.eth.getBalance(
+      DynamicGenericFeeHandlerEVMInstance.address
+    );
+
+    assert.equal(balanceAfter, Ethers.BigNumber.from(fee.toString()).add(balanceBefore).toString());
+  });
+
+  it("deposit should revert if invalid fee (msg.value) amount supplied", async () => {
+    const oracleResponse = {
+      ber: Ethers.utils.parseEther("0.000533"),
+      ter: Ethers.utils.parseEther("1.63934"),
+      dstGasPrice: Ethers.utils.parseUnits("30000000000", "wei"),
+      expiresAt: Math.floor(new Date().valueOf() / 1000) + 500,
+      fromDomainID: originDomainID,
+      toDomainID: destinationDomainID,
+      resourceID,
+      msgGasLimit,
+    };
+
+    const feeData = Helpers.createOracleFeeData(
+      oracleResponse,
+      oracle.privateKey,
+      tokenAmount
+    );
+
+    await TruffleAssert.reverts(
+      BridgeInstance.deposit(
+        destinationDomainID,
+        resourceID,
+        depositData,
+        feeData,
+        {
+          from: depositorAddress,
+          value: Ethers.utils.parseEther("1.0"),
+        }
+      ),
+      "Incorrect fee supplied"
+    );
+  });
+
+  it("deposit should revert if fee collection fails", async () => {
+    const oracleResponse = {
+      ber: Ethers.utils.parseEther("0.000533"),
+      ter: Ethers.utils.parseEther("1.63934"),
+      dstGasPrice: Ethers.utils.parseUnits("30000000000", "wei"),
+      expiresAt: Math.floor(new Date().valueOf() / 1000) + 500,
+      fromDomainID: originDomainID,
+      toDomainID: originDomainID,
+      resourceID,
+      msgGasLimit,
+    };
+
+    const feeData = Helpers.createOracleFeeData(
+      oracleResponse,
+      oracle.privateKey,
+      tokenAmount
+    );
+
+    await TruffleAssert.reverts(
+      BridgeInstance.deposit(
+        destinationDomainID,
+        resourceID,
+        depositData,
+        feeData,
+        {
+          from: depositorAddress,
+          value: Ethers.utils.parseEther("0.5").toString(),
+        }
+      )
+    );
+  });
+
+  it("deposit should revert if not called by router on DynamicFeeHandler contract", async () => {
+    const oracleResponse = {
+      ber: Ethers.utils.parseEther("0.000533"),
+      ter: Ethers.utils.parseEther("1.63934"),
+      dstGasPrice: Ethers.utils.parseUnits("30000000000", "wei"),
+      expiresAt: Math.floor(new Date().valueOf() / 1000) + 500,
+      fromDomainID: originDomainID,
+      toDomainID: destinationDomainID,
+      resourceID,
+      msgGasLimit,
+    };
+
+    const feeData = Helpers.createOracleFeeData(
+      oracleResponse,
+      oracle.privateKey,
+      tokenAmount
+    );
+
+    await TruffleAssert.reverts(
+      DynamicGenericFeeHandlerEVMInstance.collectFee(
+        depositorAddress,
+        originDomainID,
+        destinationDomainID,
+        resourceID,
+        depositData,
+        feeData,
+        {
+          from: depositorAddress,
+          value: Ethers.utils.parseEther("0.5").toString(),
```

Comment:
All test values should be valid, except for the one that is being tested.
```suggestion
          value: Ethers.utils.parseEther("0.000036777"),
```